tests/Agent/UnitTests/Core.UnitTest/Configuration/DefaultConfigurationTests.cs

```diff
@@ -1063,6 +1063,28 @@ public string IgnoreAndExpectedMessagesSetFromLocalAndServerOverrides(bool serve
             return _defaultConfig.ExpectedErrorsConfiguration.FirstOrDefault().Key + "," + _defaultConfig.IgnoreErrorsConfiguration.FirstOrDefault().Key;
         }
 
+        //password = "XYZ" obscuring-key = "123"  encrypted-password = "aWtp"
+        //password = "XYZ" obscuring-key = "456"  encrypted-password = "bGxs"
+
+        [TestCase("ABCD", "aWtp", "123", null, ExpectedResult = "XYZ")]
+        [TestCase("ABCD", "bGxs", null, "456", ExpectedResult = "XYZ")]
+        [TestCase("ABCD", "bGxs", "123", "456", ExpectedResult = "XYZ")]
+        [TestCase("ABCD", "bGxs", null, null, ExpectedResult = "ABCD")]
+        [TestCase("ABCD", null, "123", null, ExpectedResult = "ABCD")]
+
+        public string Encrypting_Decrypting_ProxyPassword_Tests(string password, string encryptedPassword, string localConfigObscuringKey, string envObscuringKey)
+        {
+            Mock.Arrange(() => _environment.GetEnvironmentVariable("OBSCURING_KEY")).Returns(envObscuringKey);
+
+            _localConfig.service.proxy.password = password;
+            _localConfig.service.obscuringKey = localConfigObscuringKey;
+            _localConfig.service.proxy.encryptedPassword = $"{encryptedPassword}";
```

Comment:
We shouldn't need to do string interpolation here.  If there are failures when this is null, something else is wrong.

Reply:
In fact, we should test for both Empty String and NULL values.

Reply:
thanks for pointing this out, this was the leftover code from using "!obscured" prefix. This shouldn't be here.